x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/hooks.ts

```diff
@@ -35,46 +32,35 @@ export function useGetAllAssignedEventFilters(
 }
 
 export function useSearchAssignedEventFilters(
-  policyId?: string,
-  filter?: string
+  eventFiltersService: EventFiltersHttpService,
+  policyId: string,
+  options: { filter?: string; page?: number; perPage?: number }
 ): QueryObserverResult<FoundExceptionListItemSchema, ServerApiError> {
-  const http = useHttp();
-  const eventFiltersService = new EventFiltersHttpService(http);
+  const { filter, page, perPage } = options;
 
   return useQuery<FoundExceptionListItemSchema, ServerApiError>(
-    ['eventFilters', 'assigned', policyId],
+    ['eventFilters', 'assigned', 'search', policyId, options],
     () => {
-      const kuery = [
-        `((exception-list-agnostic.attributes.tags:"policy:${policyId}") OR (exception-list-agnostic.attributes.tags:"policy:all"))`,
-      ];
-
-      if (filter) {
-        const filterKuery = parseQueryFilterToKQL(filter, SEARCHABLE_FIELDS) || undefined;
-        if (filterKuery) {
-          kuery.push(filterKuery);
-        }
-      }
-
       return eventFiltersService.getList({
-        filter: kuery.join(' AND '),
+        filter: parsePoliciesAndFilterToKql({
+          policies: [policyId, 'all'],
+          kuery: parseQueryFilterToKQL(filter || '', SEARCHABLE_FIELDS),
+        }),
+        perPage,
+        page: (page ?? 0) + 1,
       });
     },
     {
       refetchIntervalInBackground: false,
       refetchOnWindowFocus: false,
-      enabled: !!policyId,
-      refetchOnMount: true,
+      cacheTime: 0, // No cache this query as the search results need to be fresh
     }
   );
 }
 
-export function useGetAllEventFilters(): QueryObserverResult<
-  FoundExceptionListItemSchema,
-  ServerApiError
-> {
-  const http = useHttp();
-  const eventFiltersService = new EventFiltersHttpService(http);
-
+export function useGetAllEventFilters(
+  eventFiltersService: EventFiltersHttpService
```

Comment:
what if instead of expecting `eventFiltersService` to come as a parameter you create a custom hook that gives you the service? instead of having to create it, memoize it and pass it anywhere you want to use these hooks

Reply:
great idea!!